api/app/signals/apps/email_integrations/tests/test_rules.py

```diff
@@ -0,0 +1,220 @@
+    SignalReopenedRule,
+    SignalScheduledRule
+)
+from signals.apps.questionnaires.app_settings import NO_REACTION_RECEIVED_TEXT
+from signals.apps.signals import workflow
+from signals.apps.signals.factories import SignalFactory, StatusFactory
+
+
+class RuleTestMixin:
+    rule = None
+    state = None
+
+    def test_happy_flow(self):
+        status_text = FuzzyText(length=200) if self.state == workflow.REACTIE_GEVRAAGD else FuzzyText(length=400)
+        signal = SignalFactory.create(status__state=self.state, status__text=status_text,
+                                      reporter__email='test@example.com')
+        self.assertTrue(self.rule(signal))
+
+    def test_signal_not_happy_flow(self):
+        signal = SignalFactory.create(status__state=workflow.BEHANDELING, reporter__email='test@example.com')
+        self.assertFalse(self.rule(signal))
+
+    def test_anonymous_reporter(self):
+        status_text = FuzzyText(length=200) if self.state == workflow.REACTIE_GEVRAAGD else FuzzyText(length=400)
+
+        signal = SignalFactory.create(status__state=self.state, status__text=status_text, reporter__email='')
+        self.assertFalse(self.rule(signal))
+
+        signal = SignalFactory.create(status__state=self.state, status__text=status_text, reporter__email=None)
+        self.assertFalse(self.rule(signal))
+
+    def test_apply_for_parent_signals(self):
+        status_text = FuzzyText(length=200) if self.state == workflow.REACTIE_GEVRAAGD else FuzzyText(length=400)
+
+        parent_signal = SignalFactory.create(status__state=self.state, status__text=status_text,
+                                             reporter__email='test@example.com')
+        SignalFactory.create(status__state=self.state, status__text=status_text, reporter__email='test@example.com',
+                             parent=parent_signal)
+
+        self.assertTrue(self.rule(parent_signal))
+
+    def test_do_not_apply_for_child_signals(self):
+        status_text = FuzzyText(length=200) if self.state == workflow.REACTIE_GEVRAAGD else FuzzyText(length=400)
+
+        parent_signal = SignalFactory.create(status__state=self.state, status__text=status_text,
+                                             reporter__email='test@example.com')
+        child_signal = SignalFactory.create(status__state=self.state, status__text=status_text,
+                                            reporter__email='test@example.com', parent=parent_signal)
+
+        self.assertFalse(self.rule(child_signal))
+
+    def test_invalid_states(self):
+        states = list(map(lambda x: x[0] and x[0] == self.state, workflow.STATUS_CHOICES))
+        for state in states:
+            signal = SignalFactory.create(status__state=state, reporter__email='test@example.com')
+            self.assertFalse(self.rule(signal))
+
+
+class TestSignalCreatedRule(RuleTestMixin, TestCase):
+    rule = SignalCreatedRule()
+    state = workflow.GEMELD
+
+    def test_signal_set_state_second_time(self):
```

Comment:
Docstring with test's intention would be nice in this module (in terms of business rules we are checking). Not a blocker.